mnefun/_mnefun.py

```diff
@@ -1368,10 +1369,10 @@ def do_preprocessing_combined(p, subjects):
             meg, eeg = 'meg' in raw, 'eeg' in raw
             picks = pick_types(raw.info, meg=meg, eeg=eeg, eog=False,
                                exclude=[])
-            assert type(p.auto_bad_reject) and type(p.auto_bad_flat) == dict
+            flat = p.auto_bad if type(p.auto_bad_flat) == dict else None
```

Comment:
right now you have `flat = p.auto_bad`, not `flat = p.auto_bad_flat`, that's where the `float` can come from

yes you need the `else` clause in there